File: week2/readability/readability.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<ctype.h>
#include<math.h>

int count_letter(string usersentence);
int count_word(string usersentence);
int count_sentence(string usersentence);

int main(void)
{
    string usersentence = get_string("Text: ");
    int letters = count_letter(usersentence);
    int words = count_word(usersentence);
    int sentences = count_sentence(usersentence);

    int score = 0.0588 * letters/words *100 - 0.296 * sentences/words *100- 15.8;
    printf("total score: %i\n",score);
}

int count_letter(string usersentence)
{
    int count = 0;
    for(int i = 0, length = strlen(usersentence);i<length;i++)
    {
        if(isalpha(usersentence[i]))
        {
            count+=1;
        }
    }
    printf("letter count: %i\n",count);
    return count;
}
int count_word(string usersentence)
{
    int count = 0;
    for(int i = 0, length = strlen(usersentence);i<length;i++)
    {
        if(isspace(usersentence[i]))
        {
            count+=1;
        }
    }
    printf("word count: %i\n",count+1);
    return count+1;
}
int count_sentence(string usersentence)
{
    int count = 0;
    for(int i = 0, length = strlen(usersentence);i<length;i++)
    {
        if(usersentence[i] == '.'||usersentence[i] == '?'||usersentence[i] == '!')
        {
            count+=1;
        }
    }
    printf("sentence count: %i\n",count);
    return count;
}

File: week6/readability.py
from cs50 import get_string

def main():
    s = get_string("Text: ")
    letters = count_letter(s)
    words = count_word(s)
    sentences = count_sentence(s)
    score = 0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8

    if score >= 16:
        print("Grade 16+")
    elif score < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(score)}")

def count_letter(s):
    count = 0
    for c in s:
        if c.isalpha():
            count += 1
    return count

def count_word(s):
    count = 1  # 至少有一個單字
    for c in s:
        if c.isspace():
            count += 1
    return count

def count_sentence(s):
    count = 0
    for c in s:
        if c in ".!?":
            count += 1
    return count

main()
